doc/developer/platform/architecture-storage.md

```diff
@@ -107,25 +107,142 @@ All commands are durably persisted and their effect survives crashes and reboots
+
+3. Create `persist` collections for the output and the identified "state".
+
+
+With this prep work done, the assembly instructions are:
+
+1. To start, determine `t` the `min` write frontier in `persist` for all outputs and all state.
+
+2. From time `t` onward, rehydrate the "state" associated with the source pipeline.
+
+3. Restart each raw source from the addresses found in the persisted reclocking collection at `t`.
+
+4. From time `t` onward, continually produce reclocked messages: messages from the raw source whose addresses are translated to time using persist's reclocking information.
+
+    It is important that this be based on reclocking information we hear back from persist, rather than what we optimistically proposed at persist.
+    This ensures that the right thing happens in the case there are two contending implementations.
+
+    Each raw source should produce exactly its outputs at times greater or equal to `t`.
+
+5. From time `t` onward, apply the source pipeline to produce at each time:
+
+    a. Changes to push at persist to update the output.
+
+    b. Changes to push at persist to update the state.
+
+These are the whole instructions.
+Clearly Step 5. is not fully specified.
+The property it must have is that two independent pipelines should produce the same output and state updates.
+One way to make this happen is to behave as if each message becomes available at its timestamp, and at each time we "immediately" retire all complete work.
+For example, the first timestamp for which we have both transaction metadata and all supporting data messages, we would output those data messages.
+Alternately, the first timestamp for which we have both transaction metadata and all supporting data messages, *and no incomplete prior transactions*, we would output those data messages.
+The specific choice of what to do depends on our goals, but one attempts to do it as if executing each timestamp one at a time in sequence, and doing all the work that is possible at each timestamp.
+
+### Outcomes
+
+The reason we follow the recipe is because many good things happen automatically.
+
+1. *Definiteness.*
+
+    We get definiteness from `persist`.
+    I don't think we are able to screw this up.
+
+2. *Fault Tolerance.*
+
+    Our choice of `t` is meant to provide a "consistent snapshot" of our source pipeline.
+    It is a moment at which we know exactly where we were in input processing (indicated by the reclocking collection), what our "state" was, and what the output corresponded to.
+    With this consistent snapshot, one can resume work and proceed as if without interruption. Multiple concurrent instances should also produce identical outputs, because of the consistent snapshot and deterministic execution (onward, from the reclocking input).
+
+3. *Conciseness.*
```

Comment:
Should this be "bounded input reliance" to match the property from the start?

Reply:
Yes probably!